lib/assertions.js

```diff
@@ -1044,59 +1044,68 @@ module.exports = function (expect) {
-                                output.nl(index > 0 ? 1 : 0).i().block(function () {
-                                    var type = diffItem.type;
-                                    if (type === 'insert') {
-                                        this.annotationBlock(function () {
-                                            var index = typeof diffItem.actualIndex !== 'undefined' ? diffItem.actualIndex : diffItem.expectedIndex;
-                                            if (expect.findTypeOf(diffItem.value).is('function')) {
-                                                this.error('missing: ')
-                                                    .property(index, output.clone().block(function () {
-                                                        this.omitSubject = undefined;
-                                                        var promise = keyPromises[diffItem.expectedIndex];
-                                                        if (promise.isRejected()) {
-                                                            this.appendErrorMessage(promise.reason());
-                                                        } else {
-                                                            this.appendInspected(diffItem.value);
-                                                        }
-                                                    }), true);
-                                            } else {
-                                                this.error('missing ').property(index, inspect(diffItem.value), true);
-                                            }
-                                        });
-                                    } else {
-                                        this.property(diffItem.actualIndex, output.clone().block(function () {
-                                            if (type === 'remove') {
-                                                this.append(inspect(diffItem.value).amend(delimiterOutput.sp()).error('// should be removed'));
-                                            } else if (type === 'equal') {
-                                                this.append(inspect(diffItem.value).amend(delimiterOutput));
-                                            } else {
-                                                var toSatisfyResult = toSatisfyMatrix[diffItem.actualIndex][diffItem.expectedIndex];
-                                                var valueDiff = toSatisfyResult && toSatisfyResult !== true && toSatisfyResult.getDiff({ output: output.clone() });
-                                                if (valueDiff && valueDiff.inline) {
-                                                    this.append(valueDiff.amend(delimiterOutput));
+                                var type = diffItem.type;
+                                if (type === 'moveTarget') {
+                                    return output.clone();
```

Comment:
It feels like there should be a magicpen construct to map over an array of items and produce output for each item. I know you are just cloning the output, but it feels a bit unsafe that you might append to the outside output by accident. This is just thoughts on how we might refactor later.